src/power-card.ts

```diff
@@ -0,0 +1,148 @@
+    handleBurnt(from, tokenAddress, tokenId, value);
+  } else {
+    erc1155TokenBalanceFrom.amount = erc1155TokenBalanceFrom.amount.minus(value);
+    erc1155TokenBalanceTo.amount = erc1155TokenBalanceTo.amount.plus(value);
+
+    erc1155TokenBalanceFrom.save();
+    erc1155TokenBalanceTo.save();
+  }
+
+  createERC1155Transfer(from, to, tokenAddress, tokenId, value, event, index);
+}
+
+function handleBurnt(owner: Account, tokenAddress: Address, tokenId: BigInt, value: BigInt): void {
+  const erc1155Token = loadERC1155Token(tokenAddress, tokenId)
+  const erc1155TokenBalance = loadERC1155TokenBalance(tokenAddress, tokenId, owner)
+
+  erc1155Token.totalSupply = erc1155Token.totalSupply.minus(value)
+  erc1155Token.totalBurnt = erc1155Token.totalBurnt.plus(value)
+
+  erc1155TokenBalance.amount = erc1155TokenBalance.amount.minus(value);
+  erc1155TokenBalance.burnt = erc1155TokenBalance.amount.plus(value);
+
+  erc1155Token.save()
+  erc1155TokenBalance.save();
+}
+
+function handleMinted(owner: Account, tokenAddress: Address, tokenId: BigInt, value: BigInt): void {
+  const erc1155Token = loadERC1155Token(tokenAddress, tokenId)
+  const erc1155TokenBalance = loadERC1155TokenBalance(tokenAddress, tokenId, owner)
+
+  erc1155Token.totalSupply = erc1155Token.totalSupply.plus(value)
+
+  erc1155TokenBalance.amount = erc1155TokenBalance.amount.plus(value);
+
+  erc1155Token.save()
+  erc1155TokenBalance.save();
+}
+
+function loadERC1155TokenBalance(tokenAddress: Address, tokenId: BigInt, owner: Account): ERC1155TokenBalance {
+  let erc1155TokenBalanceId = tokenAddress.concatI32(tokenId.toI32()).concat(Address.fromHexString(owner.id));
+  let erc1155TokenBalance = ERC1155TokenBalance.load(erc1155TokenBalanceId);
+  if (erc1155TokenBalance === null) {
+    erc1155TokenBalance = new ERC1155TokenBalance(erc1155TokenBalanceId)
+
+    erc1155TokenBalance.owner = owner.id
+    erc1155TokenBalance.token = loadERC1155Token(tokenAddress, tokenId).id
+    erc1155TokenBalance.amount = ZERO_BI
+    erc1155TokenBalance.burnt = ZERO_BI
+
+    erc1155TokenBalance.save()
+  }
+
+  return erc1155TokenBalance;
+}
+
+function loadERC1155Token(tokenAddress: Address, tokenId: BigInt): ERC1155Token {
+  let erc1155TokenId = tokenAddress.concatI32(tokenId.toI32());
+
+  log.info(erc1155TokenId.toHexString(), [])
+
+  let erc1155Token = ERC1155Token.load(erc1155TokenId);
+
+  if (erc1155Token === null) {
+    erc1155Token = new ERC1155Token(erc1155TokenId)
+
+    erc1155Token.address = tokenAddress.toHexString()
+    erc1155Token.tokenId = tokenId
+    erc1155Token.totalSupply = ZERO_BI
+    erc1155Token.totalBurnt = ZERO_BI
+    erc1155Token.tokenUri = ''
+
+    erc1155Token.save()
+  }
+
+  return erc1155Token;
+}
+
+function createERC1155Transfer(from: Account, to: Account, tokenAddress: Address, tokenId: BigInt, value: BigInt, event: ethereum.Event, index: i32): void {
+  const transferId = event.transaction.hash.concatI32(index);
```

Comment:
https://github.com/overlay-market/v1-subgraph/blob/3a6d9b59cd30fca28fe10cc6b3f71b7bed626f4f/src/erc20.ts#L27

add log index for uniqueness